### ISSUE TITLE
Use run from package by default on Linux App Service Plan

### ISSUE THREAD

**Issue by asavaritayal** [Azure/azure-functions-core-tools#1069]

**Comment by ankitkumarr** [comment]
@asavaritayal is this supported for certain? @ahmelsayed I thought synctriggers would not work for Linux App Service and so Run From Package wouldn't work?

**Comment by ahmelsayed** [comment]
They are not really related. With ANT80, dedicated Linux should support run from package. We probably still shouldn't call synctriggers. Though I'm a bit unclear if premium needs sync triggers

**Comment by ankitkumarr** [comment]
Oh, I see. I tried to create an App Service plan function app and published using `WEBSITE_RUN_FROM_PACKAGE`. The functions didn't seem to show up.

**Comment by asavaritayal** [comment]
@ahmedelnably flagging this as P0 as it takes a really long time today to publish an app to Linux Dedicated. The scenario also impacts Functions on Linux Premium.

**Comment by ankitkumarr** [comment]
I investigated why this does not currently work properly. 

I found two issues here:

1. If the `WEBSITE_RUN_FROM_PACKAGE` is off, `core-tools` turns it on and start a kudulite build. However, the setting update takes a couple of minutes. So, this makes kudu not pick up the setting yet. Kudu ends up building as if `WEBSITE_RUN_FROM_PACKAGE` is off, while platform expects the filesystem as if `WEBSITE_RUN_FROM_PACKAGE` is on. This causes things to break.

2. This needs a fix in kudulite's `/api/zipdeploy` endpoint. Kudulite currently sets `siteversion.txt` under `/home/data/SitePackages` to point to a zip file, when using `WEBSITE_RUN_FROM_PACKAGE`. 
However, this seems to not be picked up, and portal does not show function apps. When I manually rename that to `packagename.txt` as suggested here https://github.com/Azure/app-service-announcements/issues/84, functions did seem to appear. I think kudulite needs to set `packagename.txt` for function apps.

For 2., I will reproduce and confirm this. If that's the case, I will make a PR to kudulite..
For 1, I talked to @ahmelsayed, and there's not much we can do other than wait for kudulite to get updated (a couple of mins). This will only happen the first time users switch to `WEBSITE_RUN_FROM_PACKAGE`.

**Comment by ankitkumarr** [comment]
For 2, kudulite PR- https://github.com/Azure-App-Service/KuduLite/pull/37

**Comment by ankitkumarr** [comment]
For 1, what if we add a querystring in kudulite, say `/api/zipdeploy?runfrompackage=true`, similar to what Ahmed proposed to `?dobuild=true` on a separate issue. 
So, we would do two things from core-tools-
- Update app settings with `WEBSITE_RUN_FROM_PACKAGE=1`.
- Call zipdeploy api with the querystring to let kudulite know to perform runfrompackage type of deployment.

This way kudulite can get the signal to perform a run from package right away. It will prepare the artifacts as if run from package is set. And then, the app settings will get updated eventually, restarting kudulite and the app to pick up those new artifacts.

@ahmelsayed, thoughts?

**Comment by ankitkumarr** [comment]
Discussed offline, for 1, it is indeed the best approach to simply wait for kudulite to update.
I made the PR here- https://github.com/Azure/azure-functions-core-tools/pull/1329

Once this PR is merged and kudulite changes are deployed (ANT83), we will be able to turn on Run From Package for Linux dedicated. 

There's nothing else to be done here yet. Marking this as __blocked on platform update__.

**Comment by fabiocav** [comment]
Assigning this back to Triaged until unblocked.
